lib/guardian/topic_guardian.rb

```diff
@@ -199,6 +198,47 @@ def can_see_deleted_topics?(category)
     is_staff? || is_category_group_moderator?(category)
   end
 
+  # Accepts an array of `Topic#id` and returns an array of `Topic#id` which the user can see.
+  def can_see_topic_ids(topic_ids: [], hide_deleted: true)
+    return topic_ids if is_admin?
+    return [] if topic_ids.compact.blank?
```

Comment:
We should probably do `topic_ids = topic_ids.compact` before this line, that way we don't end up with nil topic ids anywhere else here.